lib/hydranorth/raw_fedora.rb

```diff
@@ -0,0 +1,39 @@
+module Hydranorth
+  module RawFedora
+    
+    def get(id, subresource = '', additional_arguments = {})
+      response = perform :get, to_pair_tree(id) + subresource, additional_arguments
+    
+      xml = Nokogiri::XML.parse(response)
+      return response.code, xml
+    end
+    
+    def perform(action, path, arg_hash = {}, request_headers = {})
+      request_headers.reverse_merge!({cache_control: :no_cache})
+      user = ActiveFedora.config.credentials[:user]
+      password = ActiveFedora.config.credentials[:password]
+      endpoint = ActiveFedora.config.credentials[:url] + ActiveFedora.config.credentials[:base_path]
+      
+      url = endpoint + path + stringify_args(arg_hash)
+      
+      RestClient::Request.execute(
+        method: action, 
+        url: url,
+        user: user,
+        password: password,
+        headers: request_headers)
+    end
+    
+    def to_pair_tree(id)
+      "/#{ActiveFedora::Noid.treeify(id).strip}/"
+    end
+    
+    def stringify_args(arg_hash)
+      args = arg_hash.empty? ? '' : '?'
+      arg_hash.each { |k, v| args << k.to_s + '=' + v.to_s }
+      args
+    end
+    
+    module_function :get, :perform, :to_pair_tree, :stringify_args
```

Comment:
When you declare a method in a Module, it's an instance method. So,


```ruby
class Foo
  include Hydranorth::RawFedora
end
```
makes the methods available to *instances* of Foo, but you can't call them directly on the class or module. ```module_function``` declares Module-level aliases for the methods, so that you can call ```Hydranorth::RawFedora#get``` etc directly on the module.

You could declare them as module-level functions directly, but then they wouldn't become instance methods when mixed in to classes. It's a best-of-both-worlds thing. http://apidock.com/ruby/Module/module_function